environment/src/test/java/com/sequenceiq/environment/environment/service/DistroxServiceTest.java

```diff
@@ -0,0 +1,94 @@
+package com.sequenceiq.environment.environment.service;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import com.dyngr.core.AttemptResults;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.StackViewV4Response;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.StackViewV4Responses;
+import com.sequenceiq.distrox.api.v1.distrox.endpoint.DistroXV1Endpoint;
+
+@ExtendWith(MockitoExtension.class)
+public class DistroxServiceTest {
+
+    @Mock
+    private DistroXV1Endpoint distroXV1Endpoint;
+
+    @Mock
+    private PollerCollection pollerCollection;
+
+    @InjectMocks
+    private DistroxService underTest;
+
+    @BeforeEach
+    public void setup() {
+        ReflectionTestUtils.setField(underTest, "attempt", 5);
+        ReflectionTestUtils.setField(underTest, "sleeptime", 1);
+    }
+
+    @Test
+    public void testStopAttachedDistroxWhenNoAttachedDistroX() {
+        when(distroXV1Endpoint.list("envName", null)).thenReturn(new StackViewV4Responses(Collections.emptySet()));
+        when(pollerCollection.stopDistroXPoller(new ArrayList<>(), 1L)).thenReturn(() -> AttemptResults.finishWith(null));
+
+        underTest.stopAttachedDistrox(1L, "envName");
+
+        verify(distroXV1Endpoint, times(0)).putStopByCrns(List.of("crn"));
```

Comment:
for correct checking it should be `verify(distroXV1Endpoint, times(0)).putStopByCrns(anyListOf(String.class)));` or even better, any()

Reply:
what is the problem with the exact string? I want to run with this string only.

Reply:
exact evaluations are always better, if the author has the energy to write the test like that